File: src/_score/score_module.php
<?php

namespace Materia;

class Score_Modules_ProofReadingSymbols extends Score_Module
{

	public function check_answer($log)
	{
		if (isset($this->questions[$log->item_id]))
		{
			$q = $this->questions[$log->item_id];
			foreach ($q->answers as $answer)
			{
				if (trim($log->text) == trim($answer['text'])) return $answer['value'];
			}
		}

		return 0;
	}

	protected function details_for_question_answered($log)
	{
		// old version of qset uses default details_for_question_answered function
		if (isset($this->questions[$log->item_id])){
			return parent::details_for_question_answered($log);
		}

		// new version uses responses in the log
		$score = $log->value === $log->text ? 100 : 0;

		return [
			'data' => [
				$log->item_id,
				$log->text
			],
			'data_style'    => ['question', 'response', 'answer'],
			'score'         => $score,
			'type'          => $log->type,
			'style'         => $this->get_detail_style($score),
			'tag'           => 'div',
			'symbol'        => '%',
			'graphic'       => 'score',
			'display_score' => true
		];
	}

	protected function get_score_details()
	{
		$details = [];

		foreach ($this->logs as $log)
		{
			switch ($log->type)
			{
				case Session_Log::TYPE_QUESTION_ANSWERED:
					$details[] = $this->details_for_question_answered($log);
					break;
			}
		}

		if (count($details[0]) && count($details[0]['data']) === 2)
		{
			// new scores only have 2 data points
			$headers = ['Question Score', 'The Question', 'You Selected'];
		}
		else
		{
			// old scores have 3 data points
			$headers = $this->_ss_table_headers;
		}

		return [[
				'title'  => $this->_ss_table_title,
				'header' => $headers,
				'table'  => $details
		]];
	}
}

File: src/_score/score_module.py
from scoring.module import ScoreModule
from core.models import Log

class ProofReadingSymbols(ScoreModule):
    
    def check_answer(self, log):
        score = 100 if log.value.strip() == log.text.strip() else 0
        return score

    def details_for_question_answered(self, log):
        score = 100 if log.value.strip() == log.text.strip() else 0
        
        return {
			"data": [
				log.item_id,
				log.text
			],
			"data_style": ["question", "response", "answer"],
			"score": score,
			"type": log.log_type,
			"style": self.get_detail_style(score),
			"tag": "div",
			"symbol": "%",
			"graphic": "score",
			"display_score": True,
		}

    def get_score_details(self):
        table = []
        for log in self.logs:
            if log.log_type == Log.LogType.SCORE_QUESTION_ANSWERED:
                row = self.details_for_question_answered(log)
                table.append(row)

        if table and len(table[0].get('data', [])) == 2:
            headers = ['Question Score', 'The Question', 'You Selected']
        else:
             headers = self._ss_table_headers

        return [
            {
				"title": self._ss_table_title,
				"headers": headers,
                "table": table,
            }
        ]
